packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java

```diff
@@ -162,25 +164,43 @@ public Response createFromSimulationResult(
 	@Produces(MediaType.SERVER_SENT_EVENTS)
 	@SseElementType(MediaType.APPLICATION_JSON)
 	@Tag(name = "Stream partial/intermediate simulation result associated with run ID")
-	public Publisher<SimulationIntermediateResults> stream(
+	public Publisher<byte[]> stream(
 		@PathParam("jobId") final String jobId
 	) {
-		return Multi.createFrom().publisher(partialSimulationResults).select().where(event -> event.getJobId().equals(jobId));
+		return Multi.createFrom().publisher(partialSimulationStream).filter(event -> {
+			try{ 
+				//TODO: https://github.com/DARPA-ASKEM/Terarium/issues/1757
+				String jsonString = new String(event);
+				jsonString = jsonString.replace(" ","");
+
+				ObjectMapper mapper = new ObjectMapper();
+				SimulationIntermediateResults interResult = mapper.readValue(jsonString, SimulationIntermediateResults.class);
```

Comment:
Can we declare the mapper outside and reuse it to be more efficient? 
Or does it need to be a new instantiation for every single message?

Reply:
Very fair point will do